ui/selectors/selectors.js

```diff
@@ -718,6 +717,60 @@ export function getSnapRegistryData(state, snapId) {
   return snapsRegistryData ? snapsRegistryData[snapId] : null;
 }
 
+/**
+ * Find and return Snap's latest version available in registry.
+ *
+ * @param state - Redux state object.
+ * @param snapId - ID of a Snap.
+ * @returns String SemVer version.
+ */
+export function getSnapLatestVersion(state, snapId) {
+  const snapRegistryData = getSnapRegistryData(state, snapId);
+
+  if (!snapRegistryData) {
+    return null;
+  }
+
+  return Object.keys(snapRegistryData.versions).reduce((latest, version) => {
+    return semver.gt(version, latest) ? version : latest;
+  }, '0.0.0');
+}
+
+/**
+ * Return a Map of all installed Snaps with available update status.
+ *
+ * @param state - Redux state object.
+ * @returns Map Snap IDs mapped to a boolean value (true if update is available, false otherwise).
+ */
+export function getAllSnapsAvailableUpdateStatus(state) {
+  const snapMap = new Map();
+  const installedSnaps = state.metamask.snaps;
+
+  Object.keys(installedSnaps).forEach((snapId) => {
+    const latestVersion = getSnapLatestVersion(state, snapId);
+
+    snapMap.set(
+      snapId,
+      latestVersion
+        ? semver.gt(latestVersion, installedSnaps[snapId].version)
+        : false,
+    );
+  });
+
+  return snapMap;
+}
+
+/**
+ * Return status of Snaps update availability for any installed Snap.
+ *
+ * @param state - Redux state object.
+ * @returns boolean true if update is available, false otherwise.
+ */
+export function getSnapsUpdateAvailableStatus(state) {
+  const snapMap = getAllSnapsAvailableUpdateStatus(state);
+  return [...snapMap.values()].some((value) => value === true);
```

Comment:
Do we need to spread here?

Reply:
I think yes if I want to use `.some`, I need a regular array returned with the values from map 🤔